web/src/typeahead_helper.ts

```diff
@@ -650,14 +658,30 @@ export function sort_streams(matches: StreamPillData[], query: string): StreamPi
     return [...name_results.matches, ...desc_results.matches, ...desc_results.rest];
 }
 
-export function query_matches_person(query: string, person: User): boolean {
-    return (
-        typeahead.query_matches_string_in_order(query, person.full_name, " ") ||
-        (Boolean(person.delivery_email) &&
-            typeahead.query_matches_string_in_order(query, people.get_visible_email(person), " "))
-    );
+export function query_matches_person(
+    query: string,
+    person: UserPillData | UserOrMentionPillData,
+): boolean {
+    if (typeahead.query_matches_string_in_order(query, person.full_name, " ")) {
+        return true;
+    }
+    // TODO(evy) look into removing the delivery email check
+    if (
+        (person.type === "user" || person.is_broadcast === undefined) &&
+        Boolean(person.delivery_email)
```

Comment:
I'd like to remove this `Boolean` check and just check email for every `User` but that actually changes the behavior of this slightly and I'm not yet sure if that's okay.

Reply:
https://chat.zulip.org/#narrow/stream/6-frontend/topic/query_matches_person.20delivery.20email